### PR TITLE
Review all qualification documents regarding rustfmt qualification

### PR DESCRIPTION
**Pull request by Urhengulas**
This PR
- adapts the language to having more than one qualified tool; mostly:
    - say Ferrocene or toolchain when meaning all tools
    - say rustc when meaning the compiler
- lifts some chapters of the safety-manual from the rustc section to the general section in the beginning
- some random documentation fixes

### CODE REVIEW

File: ferrocene/doc/user-manual/src/overview.rst [hunk 1]
```diff
@@ -35,7 +39,7 @@ Further Relevant Documentation
 The Ferrocene documentation package includes:
 
 * The :doc:`specification:index`, which describes the expected behavior of Rust
-  as implemented by the Ferrocene compiler.
+  as implemented by the rustc compiler.
```
Comment (tshepang) [inline ferrocene/doc/user-manual/src/overview.rst:42]: would just leave line 42 out, else it reads like "describes behavior of Rust as implemented by Rust"

File: ferrocene/doc/user-manual/src/self-test/error-codes.rst [hunk 2]
```diff
@@ -71,16 +70,16 @@ sysroot.
 FST_007: Binary version mismatch
 --------------------------------
 
-This error occurs when the version of the ``ferrocene-self-test`` binary and
-that of the Ferrocene compiler are not the same.
+This error occurs when the versions of the ``ferrocene-self-test`` binary and
```
Comment (tshepang) [inline ferrocene/doc/user-manual/src/self-test/error-codes.rst:73]:
```suggestion
This error occurs when the version of the ``ferrocene-self-test`` binary and
```

File: ferrocene/doc/user-manual/src/self-test/error-codes.rst [hunk 3]
```diff
@@ -71,16 +70,16 @@ sysroot.
...
+that of the rustc compiler are not the same.
```
Comment (tshepang) [inline ferrocene/doc/user-manual/src/self-test/error-codes.rst:74]: you can either say *the Rust compiler* or *rustc*, generally
```suggestion
that of ``rustc`` are not the same.
```

Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/self-test/error-codes.rst:74]: Can do that. I'd go for rustc.

File: ferrocene/doc/user-manual/src/overview.rst [hunk 4]
```diff
@@ -5,8 +5,12 @@ Overview
...
 
 This Manual describes the use of Ferrocene |ferrocene_version|, the |iso_ref|
-and |iec_ref| qualified version of the Rust toolchain. Ferrocene is based on rustc
-version |rust_version|.
+and |iec_ref| qualified version of the Rust toolchain.
+
+Ferrocene is based on:
+
+* rustc version |rust_version|.
+* rustfmt version |rustfmt_version|.
```
Comment (Urhengulas) [inline ferrocene/doc/user-manual/src/overview.rst:13]: Should or must we mention the rustfmt version here?

File: ferrocene/doc/user-manual/src/overview.rst (hunk 1 above)
Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/overview.rst:42]: What would you think about following?

> the expected behavior of Rust as implemented by rustc.

File: ferrocene/doc/user-manual/src/targets/index.rst [hunk 5]
```diff
@@ -41,19 +41,16 @@ target for every code change.
      - Triple
      - Kind
      - Standard library
-     - Notes
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/targets/index.rst:44]: Let's not remove the notes column here.

File: ferrocene/doc/evaluation-plan/src/method.rst [hunk 6]
```diff
@@ -103,8 +103,6 @@ HazOp analysis:
      - Possible effects of the potential error.
    * - Mitigation 
      - The detection or mitigation method identifier, if available.
-   * - Detectable
-     - YES or NO, indicates if the error could be detected.
```
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/evaluation-plan/src/method.rst:107]: Because it is not included in the hazop for rustc: https://public-docs.ferrocene.dev/main/qualification/evaluation-report/rustc/tool-analysis.html

File: ferrocene/doc/user-manual/src/overview.rst (hunk 4 above)
Comment (pietroalbini) [reply ferrocene/doc/user-manual/src/overview.rst:13]: Yes.

Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/overview.rst:13]: Okies. Then I am gonna add the dynamic substitution.

File: ferrocene/doc/index/index.html [hunk 7]
```diff
@@ -158,6 +158,10 @@ <h3>The Reference</h3>
                         <h3>Compiler error codes</h3>
                         <p>List of error codes the compiler can emit, with additional details on how to fix them.</p>
                     </a>
+                    <a href="rustfmt/index.html">
+                        <h3>rustfmt documentation</h3>
+                        <p>Documentation about the rustfmt formatter.</p>
+                    </a>
```
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/index/index.html:164]: My plan was to point it at the rustfmt README, which is the most complete documentation there is, I think. But we can also just not link it from there.

[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/index/index.html]: Removed it

File: ferrocene/doc/user-manual/src/self-test/error-codes.rst [hunk 8]
```diff
@@ -72,15 +71,15 @@ FST_007: Binary version mismatch
...
+that of rustc are not the same.
 
 **Suggested fixes**
 
 Do not use the same ``ferrocene-self-test`` binary with different Ferrocene
 toolchains.
 
 Reinstall the Ferrocene toolchain in order to obtain the expected
-``ferrocene-self-test`` binary or Ferrocene compiler.
+``ferrocene-self-test`` binary or rustc.
```
Comment (tshepang) [inline ferrocene/doc/user-manual/src/self-test/error-codes.rst:82]: referring to a command/executable, so feels better to quote it (and also for consistency, since ferrocene-self-test command is also quoted)
```suggestion
``ferrocene-self-test`` binary or ``rustc``.
```

File: ferrocene/doc/evaluation-plan/src/method.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (pietroalbini) [reply ferrocene/doc/evaluation-plan/src/method.rst:107]: Then we should also add it to the rustc hazop, in a separate PR.

**Comment by pietroalbini** [comment]
Documentation changes look good to me (modulo Tshepang's feedback), but the code to fetch the rustfmt version should be changed as we discussed in Zulip.

File: ferrocene/doc/user-manual/src/overview.rst (hunk 4 above)
Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/overview.rst:13]: Dynamic substitution will be implemented in a follow-up PR

File: ferrocene/doc/evaluation-plan/src/method.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/evaluation-plan/src/method.rst:107]: Do you think we really need it? If we could qualify rustc without it maybe it is fine without?

File: ferrocene/doc/qualification-plan/src/details.rst [hunk 9]
```diff
@@ -4,17 +4,22 @@
 Ferrocene Details
 =================
 
-The Ferrocene toolchain consists of the ``rustc`` compiler driver, which
-internally invokes the Rust Front-End, LLVM, and LLD, in order to produce a
-library or an executable from an input Rust program.
+The Ferrocene toolchain consists of multiple tools. Following a list of the qualified ones.
+
+rustc
+-----
```
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/qualification-plan/src/details.rst:10]: You are right, I try to follow the convention the sphinx docs mentions: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

File: ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst [hunk 10]
```diff
@@ -55,12 +57,17 @@ Moreover, this analysis shows us that the likelihood of detecting these
...
 
-rustfmt provides a tool to format Rust code to conform with automotive
-[|iso_ref|] TCL 1/ASIL D level and industrial [|iec_ref|] class T3.
+rustfmt is a industrial [|iec_ref|] class T3 tool, because it generates outputs
+which can directly contribute to the executable code of the safety-related
+system.
+
+Nevertheless, we need no qualification methods, because the detection measures
+and usage restrictions listed above give us high confidence to detect potential
+errors.
```
Comment (Urhengulas) [inline ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst:73]: I reworded the IEC tool classification as we discussed with Nicole. What do you think?

PS. I also adapted the ISO tool classification and also adapted that part for rustc to stay consistent.

File: ferrocene/doc/evaluation-plan/src/qualification-scope.rst [hunk 11]
```diff
@@ -5,24 +5,29 @@ Qualification Scope
...
 qualification. It is the end-user responsibility to qualify these libraries if
 they are used in their code.
 
-The other development tools besides ``rustc`` are not qualified and are
-distributed for convenience only. This includes but is not limited to:
+There are other development tools besides ``rustc``.
+
+The qualified tools are:
+
+* ``rustfmt``
```
Comment (pietroalbini) [inline ferrocene/doc/evaluation-plan/src/qualification-scope.rst:22]: I would probably remove the first sentence, and say the qualified tools are `rustc` and `rustfmt`.

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst [hunk 12]
```diff
@@ -276,22 +276,22 @@ Moreover, this analysis shows us that the likelihood of detecting these
 potential errors is very low. Therefore, the tool error detection class is
 **TD3**.
 
-Using clause 11.4.5.4 in part 8 of the [|iso_ref|] standard, we can conclude that in
-the worst case the Tool Classification Level is **TCL3** and therefore we choose
-the following qualification methods:
+Using clause 11.4.5.4 in part 8 of the [|iso_ref|] standard, we can conclude
+that in the worst case the Tool Classification Level is **TCL3/ASIL D** and
```
Comment (pietroalbini) [inline ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:280]: Why did you change the classifcation level to be `TCL3/ASIL D`? The TCL is just `TCL3`.

File: ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst [hunk 13]
```diff
@@ -55,12 +57,17 @@ Moreover, this analysis shows us that the likelihood of detecting these
 potential errors is very high. Therefore, the tool error detection class is
 **TD1**.
 
-Using clause 11.4.5.4 in part 8 of the [|iso_ref|] standard, we can conclude that in
-the worst case the Tool Classification Level is **TCL1**. Therefore we need no
-qualification methods.
+Using clause 11.4.5.4 in part 8 of the [|iso_ref|] standard, we can conclude
+that in the worst case the Tool Classification Level is **TCL1/ASIL D**.
```
Comment (pietroalbini) [inline ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst:61]: Why did you change the classifcation level to be TCL3/ASIL D? The TCL is just TCL3.

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst (hunk 10 above)
Comment (pietroalbini) [reply ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst:73]: Looks good!

File: ferrocene/doc/evaluation-plan/src/method.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/evaluation-plan/src/method.rst:107]: Ok! Will revert that change and add a task to our clickup board

File: ferrocene/doc/evaluation-plan/src/qualification-scope.rst (hunk 11 above)
Comment (Urhengulas) [reply ferrocene/doc/evaluation-plan/src/qualification-scope.rst:22]: By first sentence you mean following?

> There are other development tools besides ``rustc``.

Comment (pietroalbini) [reply ferrocene/doc/evaluation-plan/src/qualification-scope.rst:22]: Yes.

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst (hunk 12 above)
Comment (Urhengulas) [reply ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:280]: Because before the situation was like following

- In the "Tool Evaluation Results" section we said `"the worst case the Tool Classification Level is **TCL3**"`
- In the "IEC 61508 Tool Classification" section we said `"to conform with automotive [|iso_ref|] TCL 3/ASIL D level"`

To me it does not make sense that we talk about ISO in the IEC section. Therefore I removed that mention from there. But since in the ISO section (Before "Tool Evaluation Result", now "ISO 26262 Tool Classification") we did not mention the ASIL D, which was mentioned in the IEC section, I added it there.

File: ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst (hunk 13 above)
Comment (Urhengulas) [reply ferrocene/doc/evaluation-report/src/rustfmt/tool-analysis.rst:61]: Same reasoning as above.

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst (hunk 12 above)
Comment (pietroalbini) [reply ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:280]: Oh I see!

I think saying "tool classification level" of "TCL3/ASIL D" is wrong, as ASIL D is not the tool classification level. I'd change it to be just TCL3.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Comment by Urhengulas** [comment]
Fixed CI

**Review by pietroalbini — APPROVED** [review]
bors merge
